packages/runtime/runtime-utils/src/test/requestParser.spec.ts

```diff
@@ -47,20 +48,57 @@ describe("RequestParser", () => {
             requestParser = RequestParser.create({ url: "/dataStoreId//some/route/" });
         });
         it("Create request from part 0", () => {
-            assert.equal(requestParser.createSubRequest(0)?.url, "dataStoreId/some/route");
+            assert.strictEqual(requestParser.createSubRequest(0)?.url, "/dataStoreId/some/route");
         });
         it("Create request from part 1", () => {
-            assert.equal(requestParser.createSubRequest(1)?.url, "some/route");
+            assert.strictEqual(requestParser.createSubRequest(1)?.url, "/some/route");
         });
         it("Create request from part 2", () => {
-            assert.equal(requestParser.createSubRequest(2)?.url, "route");
+            assert.strictEqual(requestParser.createSubRequest(2)?.url, "/route");
         });
         it("Create request from parts length ", () => {
-            assert.equal(requestParser.createSubRequest(3)?.url, "");
+            assert.strictEqual(requestParser.createSubRequest(3)?.url, "/");
         });
         it("Create request from invalid part ", () => {
             assert.throws(() => requestParser.createSubRequest(4));
             assert.throws(() => requestParser.createSubRequest(-1));
         });
     });
+    describe(".createSubRequest with query params", () => {
+        let requestParser2: RequestParser;
+        beforeEach(() => {
+            requestParser2 = RequestParser.create({ url: "/dataStoreId/some/route/?queryparams" });
+        });
+        it("Create request from part 0", () => {
+            assert.strictEqual(requestParser2.createSubRequest(0)?.url, "/dataStoreId/some/route/?queryparams");
+        });
+        it("Create request from part 1", () => {
+            assert.strictEqual(requestParser2.createSubRequest(1)?.url, "/some/route/?queryparams");
+        });
+        it("Create request from part 2", () => {
+            assert.strictEqual(requestParser2.createSubRequest(2)?.url, "/route/?queryparams");
+        });
+        it("Create request from parts length", () => {
+            assert.strictEqual(requestParser2.createSubRequest(3)?.url, "/?queryparams");
+        });
+        it("Create request from invalid part ", () => {
+            assert.throws(() => requestParser2.createSubRequest(4));
+            assert.throws(() => requestParser2.createSubRequest(-1));
+        });
+    });
+    describe(".createSubRequest with special urls", () => {
+        beforeEach(() => {
+        });
+        it("Create request for `/`", () => {
+            const testRequestParser = RequestParser.create({ url: "/" });
+            assert.strictEqual(testRequestParser.createSubRequest(1)?.url, "/");
```

Comment:
I think it should fail. testRequestParser.createSubRequest(0) should return "/", (1) should throw,

Reply:
I'd add testing for (0) & (1) into next it() test as well.